src/core/class_db.cpp

```diff
@@ -362,12 +362,20 @@ void ClassDB::deinitialize(GDExtensionInitializationLevel p_level) {
 
 		internal::gdextension_interface_classdb_unregister_extension_class(internal::library, name._native_ptr());
 
-		for (auto method : cl.method_map) {
+		for (std::pair<const StringName, MethodBind *> method : cl.method_map) {
 			memdelete(method.second);
 		}
 
-		classes.erase(*i);
-		class_register_order.erase((i + 1).base());
+		classes.erase(name);
+		to_erase.insert(name);
+	}
+
+	{
+		// The following is equivalent to c++20 `std::erase_if(class_register_order, [&](const StringName& name){ return to_erase.contains(name); });`
+		std::vector<StringName>::iterator it = std::remove_if(class_register_order.begin(), class_register_order.end(), [&](const StringName &name) {
```

Comment:
```suggestion
		std::vector<StringName>::iterator it = std::remove_if(class_register_order.begin(), class_register_order.end(), [&](const StringName &p_name) {
```

Reply:
No problem. BTW is this to avoid having several different "name" in the same function?

Reply:
Function arguments are prefixed by `p_` to avoid shadowing variables generally, and we prefer it everywhere